pkg/operator/controller/externaldns/deployment.go

```diff
@@ -68,10 +73,28 @@ var sourceStringTable = map[operatorv1alpha1.ExternalDNSSourceType]string{
 // ensureExternalDNSDeployment ensures that the externalDNS deployment exists.
 // Returns a Boolean value indicating whether the deployment exists, a pointer to the deployment, and an error when relevant.
 func (r *reconciler) ensureExternalDNSDeployment(ctx context.Context, namespace, image string, serviceAccount *corev1.ServiceAccount, externalDNS *operatorv1alpha1.ExternalDNS) (bool, *appsv1.Deployment, error) {
+	/*r.client.Scheme().New( schema.GroupVersionKind{
+		Group:   "config.openshift.io",
+		Version: "v1",
+		Kind:    "Infrastructure",
+	})*/
+	var platformStatus *configv1.PlatformStatus
+	var err error
+	if operatorv1alpha1.IsOpenShift {
+		infraConfig := &configv1.Infrastructure{}
+		if err = r.client.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, infraConfig); err != nil {
+			return false, &appsv1.Deployment{}, fmt.Errorf("failed to get infrastructure 'config': %v", err)
+		}
+
+		platformStatus, err = util.GetPlatformStatus(r.client, infraConfig)
```

Comment:
We don't need to get the platform status at each `ensureExternalDNSDeployment` call. This data doesn't change, same as `IsOpenShift` variable which you set once at the webhook setup.
This would avoid dragging this status through all the methods (and changing their definitions too) down to `fillXXXFields` methods. Also, the status is not used at all in case of not OpenShift deployment, so all the methods definition changed only for a specific case.